inginious-autotest

```diff
@@ -0,0 +1,281 @@
+    if res != {}:
+        output[path] = res
+
+
+def test_all_files(config, client, course_factory):
+    """
+    Test each yaml file contained in the dir_path directory, with dir_path specified in the config var, as specified in
+    the test_task function
+    :param config: dict for configuration
+    :param client: backend client of type Client
+    :param course_factory: CourseFactory object
+    :return: None
+    """
+    test_output = {}
+    client_sync = ClientSync(client)
+    dir_path = config["course_directory"]
+    tasks = os.scandir(dir_path)
+    for task in tasks:
+        if task.is_dir():
+            task_dir = os.scandir(task.path)
+            names = [item.name for item in task_dir]
+            if "task.yaml" in names:  # task directory
+                if "test" in names:  # test only if the test sub directory is present
+                    test_path = os.path.join(task.path, "test")
+                    test_files = os.scandir(test_path)
+                    for yaml_file in test_files:
+                        if not yaml_file.name.startswith('.') and yaml_file.is_file():  # Exclude possible failures
+                            test_submission_yaml(client, course_factory, yaml_file.path, test_output, client_sync)
+                task_yaml_path = os.path.join(task.path, "task.yaml")
+                test_task_yaml(task_yaml_path, test_output, course_factory, task.name, os.path.split(dir_path)[1], config)
+    if test_output != {}:  # errors in task.yaml ou submission.test
+        output = json.dumps(test_output)
+        if "file" in config:
+            with open(config["file"], "w+") as json_file:
+                json_file.write(output)
+        else:
+            print("\n{}\n".format(output))
+        exit(-1)  # or 255 if exit code is on 8 bits
+
+
+if __name__ == "__main__":
+    # Arguments parsing
+    parser = argparse.ArgumentParser(description="Test the correctness of results contained in yaml files")
+    parser.add_argument("--logging", '-l', help="enables logging", action="store_true")
+    parser.add_argument("task_dir", help="Courses directory")
+    parser.add_argument("course_dir", help="Repository for the course to test")
+    parser.add_argument("-f", "--file", help="Store in the specified file in a json format")
+
+    args = parser.parse_args()
+
+    if args.logging:
+        # Init logging
+        init_logging()
+
+    default_problem_types = {
+        problem_type.get_type(): problem_type for problem_type in [DisplayableCodeProblem,
+                                                                   DisplayableCodeSingleLineProblem,
+                                                                   DisplayableFileProblem,
+                                                                   DisplayableMultipleChoiceProblem,
+                                                                   DisplayableMatchProblem]
+    }
```

Comment:
There should be a way to add new problem types. See the `--ptype` arg in inginious-agent-mcq.